evaluation/benchmarks/the_agent_company/scripts/summarise_results.py

```diff
@@ -0,0 +1,314 @@
+        if prompt_tokens > 128000:
+            cost *= 2
+        return cost
+    elif 'qwen2-72b' in model.lower():
+        # assuming hosted on Together
+        # https://www.together.ai/pricing, accessed 12/11/2024
+        return 0.0000009 * (prompt_tokens + completion_tokens)
+    elif 'qwen2p5-72b' in model.lower():
+        # assuming hosted on Together
+        # https://www.together.ai/pricing, accessed 12/14/2024
+        return 0.0000012 * (prompt_tokens + completion_tokens)
+    elif 'llama-v3p1-405b-instruct' in model.lower():
+        # assuming hosted on Fireworks AI
+        # https://fireworks.ai/pricing, accessed 12/11/2024
+        return 0.000003 * (prompt_tokens + completion_tokens)
+    elif 'llama-v3p1-70b-instruct' in model.lower():
+        # assuming hosted on Fireworks AI
+        return 0.0000009 * (prompt_tokens + completion_tokens)
+    elif 'llama-v3p3-70b-instruct' in model.lower():
+        # assuming hosted on Fireworks AI
+        return 0.0000009 * (prompt_tokens + completion_tokens)
+    elif 'amazon.nova-pro-v1:0' in model.lower():
+        # assuming hosted on Amazon Bedrock
+        # https://aws.amazon.com/bedrock/pricing/, accessed 12/11/2024
+        return 0.0000008 * prompt_tokens + 0.0000032 * completion_tokens
+    else:
+        raise ValueError(f'Unknown model: {model}')
+
+
+def analyze_eval_json_file(filepath: str) -> Tuple[int, int]:
+    """
+    Analyze a single eval JSON file and extract the total and result from final_score.
+
+    Args:
+        filepath: Path to the JSON file
+
+    Returns:
+        Tuple containing (total, result) from final_score
+    """
+    try:
+        with open(filepath, 'r') as f:
+            data = json.load(f)
+
+        final_score = data.get('final_score', {})
+        return (final_score.get('total', 0), final_score.get('result', 0))
+    except json.JSONDecodeError as e:
+        print(f'Error decoding JSON in {filepath}: {e}')
+        return (0, 0)
+    except Exception as e:
+        print(f'Error processing {filepath}: {e}')
+        return (0, 0)
+
+
+def analyze_traj_json_file(filepath: str) -> Tuple[int, float]:
+    """
+    Analyze a single trajectory JSON file and extract the steps and tokens
+    for each step. Then estimate the cost based on the tokens and the model type.
+    Note: this is assuming there's no prompt caching at all.
+    """
+    steps: int = 0
+    cost: float = 0.0
+    with open(filepath, 'r') as f:
+        data = json.load(f)
+        response_id = None
+        for action in data:
+            if 'tool_call_metadata' in action:
+                if action['tool_call_metadata']['model_response']['id'] != response_id:
+                    response_id = action['tool_call_metadata']['model_response']['id']
+                else:
+                    # openhands displays the same model response meta data multiple times, when
+                    # a single LLM call leads to multiple actions and observations.
```

Comment:
Just curious, did this happen during your runs? What LLM returned multiple tool calls?

Reply:
Yep it happened during my experiments. Just to be clear, it's not about a LLM returning multiple tool calls (I don't know if that's possible right now but that would be interesting) - it's about OpenHands (at least version 0.14.2) logging `['tool_call_metadata']['model_response']['id']` twice for every action - observation pair.

Enclosed is an example.

Not sure if this is already fixed but looks like something that could be improved. Recording `['tool_call_metadata']['model_response']['id']` under `observation` seems a redundant.

[traj_sde-move-bustub-wiki-image.json](https://github.com/user-attachments/files/18257617/traj_sde-move-bustub-wiki-image.json)

Reply:
So it's once in the action, then the second time, in the observation for that action. It hasn't been fixed. It hasn't even been reported. The whole litellm's object for a response is twice.

Your trajectory also shows that the LLM's response, that is, the actual content, is saved 3 times in the event stream, since it's in those two, plus our `content`.

Thanks! It was on my todo list somewhere to look into the tool metadata. 🫣😅